File: Algo/MinCostFlow/Path.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace MinCostFlow
{
    public class Path
    {
        private List<Vertex> verticesPath;
        private List<Edge> edgesPath;
        //private Vertex start;
        //private Vertex end;


        public List<Edge> EdgesPath { get => edgesPath; set => edgesPath = value; }
        public List<Vertex> VerticesPath { get => verticesPath; set => verticesPath = value; }

        public Path()
        {
            this.verticesPath = new List<Vertex>();
            this.edgesPath = new List<Edge>();
        }

        public void addVertexToPath(Vertex vertex)
        {
            this.verticesPath.Add(vertex);
        }

        public void addEdgeToPath(Edge edge)
        {
            this.edgesPath.Add(edge);
        }

        public void resetPath()
        {
            this.verticesPath.Clear();
            this.edgesPath.Clear();
        }
    }
}

File: opticost/src/main/java/com/project/opticost/algorithm/Path.java
package com.project.opticost.algorithm;

import java.util.ArrayList;
import java.util.List;

public class Path {
    private List<Vertex> verticesPath;
    private List<Edge> edgesPath;

    public Path()
    {
        this.verticesPath = new ArrayList<>();
        this.edgesPath = new ArrayList<>();
    }

    public void addVertexToPath(Vertex vertex)
    {
        this.verticesPath.add(vertex);
    }

    public void addEdgeToPath(Edge edge)
    {
        this.edgesPath.add(edge);
    }

    public void resetPath()
    {
        this.verticesPath.clear();
        this.edgesPath.clear();
    }

    public List<Vertex> getVerticesPath() {
        return verticesPath;
    }

    public void setVerticesPath(List<Vertex> verticesPath) {
        this.verticesPath = verticesPath;
    }

    public List<Edge> getEdgesPath() {
        return edgesPath;
    }

    public void setEdgesPath(List<Edge> edgesPath) {
        this.edgesPath = edgesPath;
    }
}
